### PR TITLE
SLVS-853 S3459 - Unassigned members should be removed

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

File: src/SonarLint.CSharp/Helpers/SyntaxHelper.cs [hunk 1]
```diff
@@ -42,5 +43,17 @@ public static ExpressionSyntax RemoveParentheses(this ExpressionSyntax expressio
             }
             return currentExpression;
         }
+
+        public static SyntaxNode GetSelfOrTopParenthesizedExpression(this SyntaxNode node)
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Helpers/SyntaxHelper.cs]: why not take an `ExpressionSyntax` here, which IMO can be the only type which will have `ParenthesizedExpressionSyntax` as a parent? This way you get rid of the cast below

File: src/SonarLint.CSharp/Rules/AssignmentInsideSubExpression.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var assignment = (AssignmentExpressionSyntax)c.Node;
 
-                    var topParenthesizedExpression = GetTopParenthesizedExpression(assignment);
+                    var topParenthesizedExpression = (ExpressionSyntax)assignment.GetSelfOrTopParenthesizedExpression();
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/AssignmentInsideSubExpression.cs]: this cast could be removed by changing `GetSelfOrTopParenthesizedExpression` to work on `ExpressionSyntax`

File: src/SonarLint.CSharp/Rules/AssignmentInsideSubExpression.cs [hunk 3]
```diff
@@ -155,18 +155,6 @@ private static bool TryGetCoalesceExpression(ExpressionSyntax expression, out Bi
             return coalesceExpression != null && coalesceExpression.IsKind(SyntaxKind.CoalesceExpression);
         }
 
-        private static ExpressionSyntax GetTopParenthesizedExpression(ExpressionSyntax expression)
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/AssignmentInsideSubExpression.cs]: ah in fact the previous version was already working on expression syntaxes

**Comment by dbolkensteyn** [comment]
LGTM

File: src/SonarLint.CSharp/Helpers/SyntaxHelper.cs (hunk 1 above)
Comment (tvsonar) [inline src/SonarLint.CSharp/Helpers/SyntaxHelper.cs]: good point.

**Pull request merged**